lib/csv/parser.rb

```diff
@@ -476,12 +476,12 @@ def prepare_strip
       end
     end
 
-    begin
+    def string_scanner_scan_accept_string
       StringScanner.new("x").scan("x")
     rescue TypeError
-      @@string_scanner_scan_accept_string = false
+      false
     else
-      @@string_scanner_scan_accept_string = true
+      true
     end
```

Comment:
Can we use constant instead of class variable to avoid multiple `StringScanner.new.scan` check?

Reply:
Yes, I see what you mean. I can make that change soon. Thanks!